governance/src/lib.rs

```diff
@@ -36,37 +16,73 @@ pub struct GovernanceState {
     pub height: BlockHeight,
 }
 
-#[async_trait]
-pub trait Governance: Send + Sync {
-    async fn create(directory: &str, height: BlockHeight) -> Result<(), Error>;
+#[derive(Clone, Debug, Serialize, Deserialize)]
+struct Vote {
+    pub agenda_hash: Hash256,
+    pub voter: PublicKey,
+    pub signature: Signature,
+}
 
-    async fn open(directory: &str) -> Result<Self, Error>
-    where
-        Self: Sized;
+pub struct Governance<N: P2PNetwork, S: Storage> {
+    pub dms: DMS<N, S>,
+}
 
-    async fn read(&self) -> Result<GovernanceState, Error>;
+impl<N: P2PNetwork, S: Storage> Governance<N, S> {
+    pub async fn create(_dms: DMS<N, S>, _height: BlockHeight) -> Result<(), Error> {
+        unimplemented!()
+    }
 
-    async fn vote(
+    pub async fn open(_dms: DMS<N, S>) -> Result<Self, Error> {
+        unimplemented!()
+    }
+
+    pub async fn read(&self) -> Result<GovernanceState, Error> {
+        unimplemented!()
+    }
+
+    pub async fn vote(
         &mut self,
         network_config: &NetworkConfig,
         known_peers: &[Peer],
         agenda_hash: Hash256,
         private_key: &PrivateKey,
-    ) -> Result<(), Error>;
+    ) -> Result<(), Error> {
+        let data = serde_json::to_string(&Vote {
+            agenda_hash,
+            voter: private_key.public_key(),
+            signature: Signature::sign(agenda_hash, private_key)?,
+        })
+        .unwrap();
+        let message = Message::new(
+            data.clone(),
+            TypedSignature::sign(&data, &network_config.private_key)?,
+        )?;
+
```

Comment:
delete empty line?

Reply:
maybe, but not important. I'll skip this.